HdfsAccessor.go

```diff
@@ -118,14 +118,17 @@ func (this *hdfsAccessorImpl) connectToNameNodeImpl(nnAddr string) (*hdfs.Client
 
 // Opens HDFS file for reading
 func (this *hdfsAccessorImpl) OpenRead(path string) (ReadSeekCloser, error) {
-	client, err1 := this.ConnectToNameNode()
-	if err1 != nil {
-		return nil, err1
+	// Blocking read. This is to reduce the connections pressue on hadoop-name-node
+	this.MetadataClientMutex.Lock()
+	defer this.MetadataClientMutex.Unlock()
+	if this.MetadataClient == nil {
```

Comment:
Probably no, if we still need MetadataClient lock. But as said in another comment, if MetadataClient lock is removed, we can wrap this check.